[RUN-3006][RUN-3003][RUN-3148][RUN-3135] Chromium getObjectPreview missing overflow + more getObjectPreview fixes

Domiii opened the pull request:
* paired w/ tests: https://github.com/replayio/backend/pull/9313
* [RUN-3006](https://linear.app/replay/issue/RUN-3006/chromium-object-missing-properties#comment-e7e739a3)
* Mini fix: [RUN-3003](https://linear.app/replay/issue/RUN-3003/weakmaps-and-weaksets-missing-containerentrycount-property#comment-b46eed58)
* FE E2E Tests: ✅
  * The relevant tests are not yet passing and RUN-3127 is blocking us here (see below).
  * But manual inspection reveals that all props are coming back correctly now, as, for example, seen [here](https://app.replay.io/recording/e2e-example-docrrobjectshtml--543526ab-b25b-4c2f-8889-be78d7e90917?point=43811017110635676236&time=10.25&focusWindow=eyJiZWdpbiI6eyJwb2ludCI6IjAiLCJ0aW1lIjowfSwiZW5kIjp7InBvaW50IjoiMzI0NTE4NTUzNjU5MzQ2NzU4MTQzODMyMjg0NDYzMTA0IiwidGltZSI6NTN9fQ%253D%253D) (which is a golden recording coming from the most recent test run).

WARNING: Blocked by FE E2E tests not working anymore ([RUN-3127](https://linear.app/replay/issue/RUN-3127/run-fe-e2e-tests-error-failed-establishing-connection-to-agent-you)). Once they do, the corresponding tests should be enabled.

klochek commented on .buildkite/pipeline.yml:
```diff
@@ -156,6 +156,8 @@ steps:
   - label: ":hammer: Trigger Runtime FE Tests"
     trigger: "runtime-fe-end-to-end-tests"
+    build:
+      branch: dominik/run-3006-chromium-object-missing-properties
```
Remove, pls.

Domiii replied:
I will once we get FE tests to give us a good run :D

toshok reviewed (commented):
I have to admit, this code took some serious reading to understand before, and it's quite a bit harder now with this change.  Not suggesting outright that there's a way to address that, but it's a PR review datapoint if nothing else :/

I think along with tests that make sure overflow is properly true, we need tests specifically for cases where paging is required to get the number of properties we want on objects with custom previewers, like `Errors`.

Domiii commented on third_party/blink/renderer/bindings/core/v8/record_replay_interface.cc:
```diff
@@ -1429,56 +1432,129 @@ ProtocolObjectPreview.prototype = {
-  get pageIndex() { return 0; },
-
-  get pageSize() {
-    if (isBlinkObject(this.raw, this.cdpObj) || CustomPreviewers[this.cdpObj.className]) {
```
1. Yes you are right. I think I added the `|| CustomPreviewers...` check a while back basically exactly because of the `pageSize` issues that now require us to loop. However that is a big performance hit, because big `Array`, `Set`, `Map` etc. are now not paged anymore. So now I thought, let's finally remove it again!
2. However... There is a concern with `Set` and `Map`, because they store their data in `internalProperties`, which we are currently not looping.
3. I thus added a check for a **_small_** `Set` [here](https://github.com/replayio/backend/pull/9313/files#diff-9bc33484e9051eb27d451a451dfb9b2094d530c8a1275d5ff8c612b9c8286065R130). I saw it pass and moved on. But you make the right point: Let's make sure this works quite right.

I'll add more tests and fix more bugs, as I should have from the beginning!

Domiii commented on third_party/blink/renderer/bindings/core/v8/record_replay_interface.cc:
```diff
@@ -1429,56 +1432,129 @@ ProtocolObjectPreview.prototype = {
-    // Add data for blink objects
-    this.extra = getExtraObjectPreviewData(this.cdpObj, cdpProperties) || {};
+    /**
+     * Note: The following logic only reads `internalProperties` from
```
You are correct.

Domiii commented:
@toshok:

* I hope I addressed all your concerns.
* I also fixed up and ran more tests (manually) from [here](https://github.com/replayio/backend/pull/9313).
* Since our FE E2E test runs are broken, I manually checked one of the golden recordings as explained + linked in the description.
